File: clients/asset-client/src/constants/CompanyInfoStatusConstant.ts
import { createConstant } from "./Constant";

const CompanyInfoStatusConstant = createConstant(
  {
    MoiTao: 0,
    DaDuyet: 1,
    BoSung: 2,
    ChinhSua: 3,
    TuChoi: 4,
    BiKhoa: 5,
    ChamDut: 6,
    ChoDuyet: 7,
    DeNghiChamDut: 8,
  } as const,
  {
    0: { displayName: "Mới tạo", color: "#8c8c8c" },
    1: { displayName: "Đã duyệt", color: "#16a34a" },
    2: { displayName: "Bổ sung", color: "#faad14" },
    3: { displayName: "Chỉnh sửa", color: "#faad14" },
    4: { displayName: "Từ chối", color: "#ff4d4f" },
    5: { displayName: "Bị khoá", color: "#ff4d4f" },
    6: { displayName: "Chấm dứt", color: "#ff4d4f" },
    7: { displayName: "Chờ duyệt", color: "#0355a2" },
    8: { displayName: "Đề nghị chấm dứt", color: "#faad14" },
  },
);

export default CompanyInfoStatusConstant;

File: services/kpi-service/Hinet.Service/Constant/CompanyInfoStatusConstant.cs
using System.ComponentModel;

namespace Hinet.Service.Constant
{
    public class CompanyInfoStatusConstant
    {
        [DisplayName("Mới tạo")]            public static int MoiTao => 0;
        [DisplayName("Đã duyệt")]           public static int DaDuyet => 1;
        [DisplayName("Bổ sung")]            public static int BoSung => 2;
        [DisplayName("Chỉnh sửa")]          public static int ChinhSua => 3;
        [DisplayName("Từ chối")]            public static int TuChoi => 4;
        [DisplayName("Bị khoá")]            public static int BiKhoa => 5;
        [DisplayName("Chấm dứt")]           public static int ChamDut => 6;
        [DisplayName("Chờ duyệt")]          public static int ChoDuyet => 7;
        [DisplayName("Đề nghị chấm dứt")]   public static int DeNghiChamDut => 8;
    }
}
